__tests__/test-db.test.js

```diff
@@ -255,20 +255,79 @@ test("should return error code 404 and error message when passed a id that is a
     });
 });
 
-describe('GET /api/users', () => {
-  test('should return a list of all users with the correct data', () => {
+describe("GET /api/users", () => {
+  test("should return a list of all users with the correct data", () => {
     return request(app)
       .get("/api/users")
       .expect(200)
       .then((result) => {
-        const body = result.body.users
+        const body = result.body.users;
         body.forEach((user) => {
           expect(user).toMatchObject({
             username: expect.any(String),
             name: expect.any(String),
-            avatar_url: expect.any(String)
-          })
-        })
-      })
+            avatar_url: expect.any(String),
+          });
+        });
+      });
   });
-});
\ No newline at end of file
+});
+
+describe("GET /api/articles(topicQuery)", () => {
+  test("accept a topic query and return the correct status with only the data related to the specified query", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then((result) => {
+        expect(result.body.articles.length).toBe(1);
+        expect(result.body.articles).toBeSortedBy("cats");
+      });
+  });
+  test("Should return an error if passed a query that is not accepted in the topic variables", () => {
+    return request(app)
+      .get("/api/articles?topic=dogs")
+      .expect(404)
+      .then((response) => {
+        const error = response.body;
+        expect(error.msg).toBe("Not found");
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id(comment_count)", () => {
+  test("should return a totoal of all the comments in an article specified by ID, add to the article as a comment_count key/value.", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((result) => {
+        const body = result.body.article;
+        expect(body).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+          comment_count: expect.any(String),
+        });
+      });
+  });
+  test("should return 400 code and correct message when passed an id that is not a number", () => {
+    return request(app)
+      .get("/api/articles/not_an_id")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("should return error code 404 and error message when passed a id that is a number but doesnt exist in the database", () => {
+    return request(app)
+      .get("/api/articles/999")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Article not found");
+      });
+  });
```

Comment:
What if it's passed an article that exists but has no associated comments? It would return an empty array just like the test above, but it wouldn't technically be an error.